core/core_test.go

```diff
@@ -34,13 +34,19 @@ func TestChallenges(t *testing.T) {
 		t.Errorf("New tls-sni-01 challenge is not sane: %v", tlssni01)
 	}
 
+	tlssni02 := TLSSNIChallenge02()
```

Comment:
Yes, I added two additional checks at the end of the test to ensure that it rejects the challenge when the feature is not enabled.